webapp/src/js/services/feedback.js

```diff
@@ -0,0 +1,100 @@
+/*
+  Feedback service
+ */
+angular.module('inboxServices').factory('Feedback',
+  function() {
+
+    'ngInject';
+    'use strict';
+
+    var levels = ['error', 'warn', 'log', 'info'],
+        log = [],
+        options = {};
+
+    var getUrl = function() {
+      var url = options.document && options.document.URL;
+      if (url) {
+        // blank out passwords
+        return url.replace(/:[^@:]*@/, ':********@');
+      }
+      return null;
+    };
+
+    var registerConsoleInterceptor = function() {
+      // intercept console logging
+      levels.forEach(function(level) {
+        var original = options.console[level];
+        options.console[level] = function() {
+          // push the error onto the stack
+          log.splice(0, 0, { level: level, arguments: JSON.stringify(arguments) });
+          // remove any old log entries
+          log.splice(20, Number.MAX_VALUE);
+          // output to the console as per usual
+          original.apply(options.console, arguments);
+        };
+      });
+    };
+
+    var create = function(info, isManual, callback) {
+      options.getUserCtx(function(err, userCtx) {
```

Comment:
Now that this is an angular service you can just include the Session dependency and use `Session.userCtx()` directly so we don't need to pass it in all the time.